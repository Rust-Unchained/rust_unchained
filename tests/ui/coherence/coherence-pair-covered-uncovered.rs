//@ aux-build:coherence_concrete_lib.rs

extern crate coherence_concrete_lib as lib;
use lib::{Remote, Pair, RemoteStruct};

struct Local<T>(T);

impl<T,U> Remote for Pair<Local<U>, T> { }
//~^ ERROR E0119

fn main() { }
