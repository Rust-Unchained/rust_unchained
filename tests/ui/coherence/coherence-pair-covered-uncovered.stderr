error[E0119]: conflicting implementations of trait `Remote` for type `Pair<Local<_>, RemoteStruct>`
  --> $DIR/coherence-pair-covered-uncovered.rs:8:1
   |
LL | impl<T,U> Remote for Pair<Local<U>, T> { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `coherence_concrete_lib`:
           - impl<T> Remote for Pair<T, RemoteStruct>;

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
